### ISSUE TITLE
undefined method `edit_version' for nil:NilClass

### ISSUE THREAD

**Issue by macbaszii** [fastlane/fastlane#8660]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

##### Complete output when running fastlane, including the stack trace and command used
> You can use: `--capture_output` as the last commandline argument to get that collected for you

```
bundle exec fastlane beta --verbose
Your Appfile has had smart quotes sanitised. To avoid issues in the future, you should not use TextEdit for editing it. If you are not using TextEdit, you should turn off smart quotes in your editor of choice.
Your Appfile has had smart quotes sanitised. To avoid issues in the future, you should not use TextEdit for editing it. If you are not using TextEdit, you should turn off smart quotes in your editor of choice.
DEBUG [2017-03-27 00:04:53.58]: Checking if there are any plugins that should be loaded...
WARN [2017-03-27 00:04:53.59]: You require a gem, if this is a third party gem, please use `fastlane_require "spaceship"` to ensure the gem is installed locally
INFO [2017-03-27 00:04:53.59]: -------------------------------------------------
INFO [2017-03-27 00:04:53.59]: --- Step: Verifying required fastlane version ---
INFO [2017-03-27 00:04:53.59]: -------------------------------------------------
INFO [2017-03-27 00:04:53.59]: Your fastlane version 2.23.0 matches the minimum requirement of 2.23.0  ✅
INFO [2017-03-27 00:04:53.60]: ------------------------------
INFO [2017-03-27 00:04:53.60]: --- Step: default_platform ---
INFO [2017-03-27 00:04:53.60]: ------------------------------
INFO [2017-03-27 00:04:53.60]: Driving the lane 'ios beta' 🚀
WARN [2017-03-27 00:04:56.99]: Lane Context:
INFO [2017-03-27 00:04:56.99]: {:DEFAULT_PLATFORM=>:ios, :PLATFORM_NAME=>:ios, :LANE_NAME=>"ios beta"}
ERROR [2017-03-27 00:04:56.99]: undefined method `edit_version' for nil:NilClass
INFO [2017-03-27 00:04:56.99]: Successfully generated documentation at path '/Users/distiller/livepop-ios/fastlane/README.md'

+------+-------------------------------------+-------------+
|                     fastlane summary                     |
+------+-------------------------------------+-------------+
| Step | Action                              | Time (in s) |
+------+-------------------------------------+-------------+
| 1    | Verifying required fastlane version | 0           |
| 2    | default_platform                    | 0           |
+------+-------------------------------------+-------------+

ERROR [2017-03-27 00:04:57.01]: fastlane finished with errors

Looking for related GitHub issues on fastlane/fastlane...
Search query: undefined method `edit_version' for nil

URL: https://api.github.com/search/issues?q=undefined%20method%20%60edit_version'%20for%20nil+repo:fastlane/fastlane
➡️  undefined method `map' for nil:NilClass
   https://github.com/fastlane/fastlane/issues/5817 [closed] 23 💬
   06 Dec 2016

➡️  NoMethodError: undefined method `update_testing_status!' for nil:NilClass
   https://github.com/fastlane/fastlane/issues/2050 [closed] 8 💬
   06 Dec 2016

➡️  Spaceship parsing fails with "undefined method `fetch' for nil:NilClass"
   https://github.com/fastlane/fastlane/issues/5701 [closed] 2 💬
   06 Dec 2016

and 4 more at: https://github.com/fastlane/fastlane/search?q=undefined%20method%20%60edit_version'%20for%20nil&type=Issues&utf8=✓

🔗  You can ⌘ + double-click on links to open them directly in your browser.
bundler: failed to load command: fastlane (/Users/distiller/livepop-ios/vendor/bundle/ruby/2.3.0/bin/fastlane)
NoMethodError: [!] undefined method `edit_version' for nil:NilClass
  Fastfile:29:in `block (2 levels) in parsing_binding'
  /Users/distiller/livepop-ios/vendor/bundle/ruby/2.3.0/gems/fastlane-2.23.0/fastlane/lib/fastlane/runner.rb:268:in `execute_flow_block'
  /Users/distiller/livepop-ios/vendor/bundle/ruby/2.3.0/gems/fastlane-2.23.0/fastlane/lib/fastlane/runner.rb:46:in `block in execute'
  /Users/distiller/livepop-ios/vendor/bundle/ruby/2.3.0/gems/fastlane-2.23.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
  /Users/distiller/livepop-ios/vendor/bundle/ruby/2.3.0/gems/fastlane-2.23.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
  /Users/distiller/livepop-ios/vendor/bundle/ruby/2.3.0/gems/fastlane-2.23.0/fastlane/lib/fastlane/lane_manager.rb:52:in `cruise_lane'
  /Users/distiller/livepop-ios/vendor/bundle/ruby/2.3.0/gems/fastlane-2.23.0/fastlane/lib/fastlane/command_line_handler.rb:30:in `handle'
  /Users/distiller/livepop-ios/vendor/bundle/ruby/2.3.0/gems/fastlane-2.23.0/fastlane/lib/fastlane/commands_generator.rb:104:in `block (2 levels) in run'
  /Users/distiller/livepop-ios/vendor/bundle/ruby/2.3.0/gems/commander-fastlane-4.4.4/lib/commander/command.rb:178:in `call'
  /Users/distiller/livepop-ios/vendor/bundle/ruby/2.3.0/gems/commander-fastlane-4.4.4/lib/commander/command.rb:153:in `run'
  /Users/distiller/livepop-ios/vendor/bundle/ruby/2.3.0/gems/commander-fastlane-4.4.4/lib/commander/runner.rb:476:in `run_active_command'
  /Users/distiller/livepop-ios/vendor/bundle/ruby/2.3.0/gems/fastlane-2.23.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:38:in `run!'
  /Users/distiller/livepop-ios/vendor/bundle/ruby/2.3.0/gems/commander-fastlane-4.4.4/lib/commander/delegates.rb:15:in `run!'
  /Users/distiller/livepop-ios/vendor/bundle/ruby/2.3.0/gems/fastlane-2.23.0/fastlane/lib/fastlane/commands_generator.rb:303:in `run'
  /Users/distiller/livepop-ios/vendor/bundle/ruby/2.3.0/gems/fastlane-2.23.0/fastlane/lib/fastlane/commands_generator.rb:42:in `start'
  /Users/distiller/livepop-ios/vendor/bundle/ruby/2.3.0/gems/fastlane-2.23.0/fastlane/lib/fastlane/cli_tools_distributor.rb:65:in `take_off'
  /Users/distiller/livepop-ios/vendor/bundle/ruby/2.3.0/gems/fastlane-2.23.0/bin/fastlane:20:in `<top (required)>'
  /Users/distiller/livepop-ios/vendor/bundle/ruby/2.3.0/bin/fastlane:23:in `load'
  /Users/distiller/livepop-ios/vendor/bundle/ruby/2.3.0/bin/fastlane:23:in `<top (required)>'

bundle exec fastlane beta --verbose returned exit code 1

Action failed: bundle exec fastlane beta --verbose
```

### Environment

Please run `fastlane env` and copy the output below. This will help us help you :+1:
If you used `--capture_output` option please remove this block - as it is already included there.

<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                 | Value                                       |
| ------------------- | ------------------------------------------- |
| OS                  | 10.12.3                                     |
| Ruby                | 2.3.1                                       |
| Bundler?            | true                                        |
| Git                 | git version 2.6.1                           |
| Installation Source | ~/.rvm/gems/ruby-2.3.1/bin/fastlane         |
| Host                | Mac OS X 10.12.3 (16D32)                    |
| Ruby Lib Dir        | ~/.rvm/rubies/ruby-2.3.1/lib                |
| OpenSSL Version     | OpenSSL 1.0.2h  3 May 2016                  |
| Is contained        | false                                       |
| Xcode Path          | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version       | 8.2.1                                       |

### System Locale

| Error                       |
| --------------------------- |
| No Locale with UTF8 found 🚫 |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# Customise this file, documentation can be found here:
# https://github.com/fastlane/fastlane/tree/master/fastlane/docs
# All available actions: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Actions.md
# can also be listed using the `fastlane actions` command

# Change the syntax highlighting to Ruby
# All lines starting with a # are ignored when running `fastlane`

# If you want to automatically update fastlane if a new version is available:
# update_fastlane

# This is the minimum version number required.
# Update this, if you use features of a newer version

require "spaceship"

fastlane_version "1.111.0"

default_platform :ios

platform :ios do
  before_all do
    # ENV["SLACK_URL"] = "https://hooks.slack.com/services/..."
    cocoapods unless Helper.is_ci?

    # Increment build number to one number greater than the current build on iTunes Connect
    # TODO: Use `train` endpoint to get the latest build instead of `edit_version`
    Spaceship::Tunes.login
    version = Spaceship::Tunes::Application.find("com.livepop.LivePop").edit_version.candidate_builds.first.build_version
    increment_build_number(build_number: version.to_i + 1, xcodeproj: "LivePop/LivePop.xcodeproj")
    increment_build_number(build_number: version.to_i + 1, xcodeproj: "LivePopKit/LivePopKit.xcodeproj")
  end

  desc "Runs all the tests"
  lane :test do
    scan
  end

  desc "Submit a new Beta Build to Apple TestFlight"
  desc "This will also make sure the profile is up to date"
  lane :beta do
    # match(type: "appstore") # more information: https://codesigning.guide
    sigh
    gym(scheme: "LivePop") # Build your app - more options available
    pilot(skip_submission: true, skip_waiting_for_build_processing: true)

    # sh "your_script.sh"
    # You can also use other beta testing services here (run `fastlane actions`)
  end

  desc "Deploy a new version to the App Store"
  lane :release do
    # match(type: "appstore")
    sigh
    # snapshot
    gym(scheme: "LivePop") # Build your app - more options available
    deliver(force: true)
    # frameit
  end

  # You can define as many lanes as you want

  after_all do |lane|
    # This block is called, only if the executed lane was successful

    # slack(
    #   message: "Successfully deployed new App Update."
    # )
  end

  error do |lane, exception|
    # slack(
    #   message: exception.message,
    #   success: false
    # )
  end
end


# More information about multiple platforms in fastlane: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Platforms.md
# All available actions: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Actions.md

# fastlane reports which actions are used
# No personal data is recorded. Learn more at https://github.com/fastlane/enhancer
```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier "com.livepop.LivePop" # The bundle identifier of your app
apple_id “support@oozou.com” # Your Apple email address

team_id "BA36Y82DMA"  # Developer Portal Team ID

# you can even provide different app identifiers, Apple IDs and team names per lane:
# More information: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Appfile.md
```
</details>

### fastlane gems

| Gem      | Version | Update-Status      |
| -------- | ------- | ------------------ |
| fastlane | 2.11.0  | 🚫 Update available |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.0.0        |
| executable-hooks          | 1.3.2        |
| bundler-unload            | 1.0.2        |
| rubygems-bundler          | 1.4.4        |
| bundler                   | 1.13.6       |
| io-console                | 0.4.5        |
| CFPropertyList            | 2.3.4        |
| i18n                      | 0.7.0        |
| json                      | 1.8.6        |
| minitest                  | 5.10.1       |
| thread_safe               | 0.3.5        |
| tzinfo                    | 1.2.2        |
| activesupport             | 4.2.7.1      |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.0        |
| babosa                    | 1.0.2        |
| claide                    | 1.0.1        |
| fuzzy_match               | 2.0.4        |
| nap                       | 1.1.0        |
| cocoapods-core            | 1.1.1        |
| cocoapods-deintegrate     | 1.0.1        |
| cocoapods-downloader      | 1.1.3        |
| cocoapods-plugins         | 1.0.0        |
| cocoapods-search          | 1.0.0        |
| cocoapods-stats           | 1.0.0        |
| netrc                     | 0.7.8        |
| cocoapods-trunk           | 1.1.2        |
| cocoapods-try             | 1.1.0        |
| colored                   | 1.2          |
| escape                    | 0.0.4        |
| fourflusher               | 2.0.1        |
| gh_inspector              | 1.0.3        |
| molinillo                 | 0.5.5        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.4.2        |
| cocoapods                 | 1.1.1        |
| highline                  | 1.7.8        |
| commander                 | 4.4.3        |
| unf_ext                   | 0.0.7.2      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20161129 |
| dotenv                    | 2.1.2        |
| excon                     | 0.54.0       |
| multipart-post            | 2.0.0        |
| faraday                   | 0.11.0       |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday_middleware        | 0.11.0.1     |
| fastimage                 | 2.0.1        |
| jwt                       | 1.5.6        |
| little-plugger            | 1.1.4        |
| multi_json                | 1.12.1       |
| logging                   | 2.1.0        |
| memoist                   | 0.15.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.3        |
| googleauth                | 0.5.1        |
| httpclient                | 2.8.3        |
| hurley                    | 0.2          |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| uber                      | 0.0.15       |
| representable             | 2.3.0        |
| retriable                 | 2.1.0        |
| google-api-client         | 0.9.23       |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| plist                     | 3.2.0        |
| rubyzip                   | 1.2.0        |
| security                  | 0.1.3        |
| slack-notifier            | 1.5.1        |
| terminal-notifier         | 1.7.1        |
| unicode-display_width     | 1.1.3        |
| terminal-table            | 1.7.3        |
| word_wrap                 | 1.0.0        |
| rouge                     | 1.11.1       |
| xcpretty                  | 0.2.4        |
| xcpretty-travis-formatter | 0.0.4        |
</details>

*generated on:* **2017-03-27**
</details>

**Comment by macbaszii** [comment]
I checked, there should be no problem.

**Comment by KrauseFx** [comment]
We've seen this error before, thanks for reporting. This is usually just a temporary server issue, can you give it another try?

**Comment by KrauseFx** [comment]
```
    version = Spaceship::Tunes::Application.find("com.livepop.LivePop").edit_version.candidate_builds.first.build_version
```

This line in your Fastfile isn't robust, as it doesn't verify that an `edit_version` is actually available. This might not be the case if the servers are unavailable.

**Comment by macbaszii** [comment]
@KrauseFx Thank you! That because the result of `Spaceship::Tunes::Application.find("com.livepop.LivePop")` is nil, right?

I tried to use the `train` api to get the build number instead.
But, the problem is come before that. 
The problem is `Spaceship` cannot find that particular application by bundle identifier.

The another thing is I tried it local on my machine, it's working well.
But, this problem is always found on CircleCI.
